platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -174,6 +175,9 @@ private void initialise(@NonNull final Context context, @NonNull final MapboxMap
     // notify Map object about current connectivity state
     nativeMapView.setReachability(ConnectivityReceiver.instance(context).isConnected(context));
 
+    // bind internal components for map change events
+    mapChangeDispatcher.bind(mapCallback = new MapChangeResultHandler(mapboxMap));
```

Comment:
Would it somehow be feasible to make this a constructor argument? Since it is required. That would make it possible to skip a lot of null checks in `MapChangeDispatcher`